imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
-
-    # create the attribute manager for this data level
+    # TODO add logging
+
+    # Unpack ccsds file
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/" "hit_packet_definitions.xml"
+    )
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=packet_file,
+        xtce_packet_definition=packet_definition,
+    )
+
+    # Create the attribute manager for this data level
     attr_mgr = ImapCdfAttributes()
     attr_mgr.add_instrument_global_attrs(instrument="hit")
     attr_mgr.add_instrument_variable_attrs(instrument="hit", level="l1a")
     attr_mgr.add_global_attribute("Data_version", data_version)
 
-    # Create datasets
-    datasets = create_datasets(grouped_data, attr_mgr)
+    # Process science to l1a.
+    processed_data = []
+    for apid in datasets_by_apid:
+        if apid == HitAPID.HIT_HSKP:
+            housekeeping_dataset = process_housekeeping(
+                datasets_by_apid[apid], attr_mgr
+            )
+            processed_data.append(housekeeping_dataset)
+        elif apid == HitAPID.HIT_SCIENCE:
+            # TODO complete science data processing
+            print("Skipping science data for now")
+            # science_dataset = process_science(datasets_by_apid[apid], attr_mgr)
+        else:
+            raise Exception(f"Unknown APID [{apid}]")
 
-    return list(datasets.values())
+    return processed_data
 
 
-def decom_packets(packet_file: str) -> list:
+def concatenate_leak_variables(dataset: xr.Dataset) -> xr.Dataset:
     """
-    Unpack and decode packets using CCSDS file and XTCE packet definitions.
+    Concatenate leak variables in the dataset.
+
+    Updates the housekeeping dataset to replace the individual
+    leak_i_00, leak_i_01, ..., leak_i_63 variables with a single
+    leak_i variable as a 2D array. This variable represents
+    leakage current [Voltage] data.
 
     Parameters
     ----------
-    packet_file : str
-        Path to the CCSDS data packet file.
+    dataset : xarray.Dataset
+        Dataset containing 64 leak variables.
 
     Returns
     -------
-    unpacked_packets : list
-        List of all the unpacked data.
+    dataset : xarray.Dataset
+        Updated dataset with concatenated leak variables.
     """
-    # TODO: update path to use a combined packets xtce file
-    xtce_file = imap_module_directory / "hit/packet_definitions/P_HIT_HSKP.xml"
-    logger.debug(f"Unpacking {packet_file} using xtce definitions in {xtce_file}")
-    unpacked_packets: list = decom.decom_packets(packet_file, xtce_file)
-    logger.debug(f"{packet_file} unpacked")
-    return unpacked_packets
+    # Stack 64 leak variables (leak_00, leak_01, ..., leak_63)
+    leak_vars = [dataset[f"leak_i_{i:02d}"] for i in range(64)]
 
+    # Concatenate along 'leak_index' and reorder dimensions
+    stacked_leaks = xr.concat(leak_vars, dim="leak_index").transpose(
+        "epoch", "leak_index"
+    )
+    dataset["leak_i"] = stacked_leaks
```

Comment:
Do you want to drop the "i" here and just call it "leak"?

Reply:
@greglucas The variable is for "leakage current", so `i` here represents "current", not an index, so I think it's useful in the variable name. Maybe this should be a capital `I` and a comment somewhere so that's clear.

Reply:
Ahh, that makes sense once you know it :) I think it is fine to keep it as is, This is just housekeeping and they can do what they want with it. Thanks for clarifying!